src/NearbyInteraction/NICompat.cs

```diff
@@ -0,0 +1,41 @@
+//
+// NICompat.cs
+//
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Foundation;
+using CoreFoundation;
+using ObjCRuntime;
+
+#if __MACCATALYST__ || !IOS
+using ARSession = Foundation.NSObject;
+#else
+using ARKit;
+#endif
+
+#if NET
+using Vector3 = global::System.Numerics.Vector3;
+using MatrixFloat4x4 = global::CoreGraphics.NMatrix4;
+#else
+using NativeHandle = System.IntPtr;
+using Vector3 = global::OpenTK.Vector3;
+using MatrixFloat4x4 = global::OpenTK.NMatrix4;
+#endif
+
+#nullable enable
+namespace NearbyInteraction {
+
+#if WATCH
+	public partial class NISession {
+
+		public virtual void SetARSession (ARSession session) => throw new InvalidOperationException (Constants.ApiRemovedGeneral);
```

Comment:
I think you need the platform supported attributes probably Obsolete?

Reply:
True, I forgot the obsolete. Good catch

Reply:
MORE!!!

![image](https://user-images.githubusercontent.com/204671/195898785-d19689ca-faf6-41de-9518-2ff210f6e2cb.png)

Reply:
Nope, this is only for watch. Watch is not present on Net, so we do not need to add anything more than the Obsolete for legacy.